### PR TITLE
Hammerdb LSO support

### PR DESCRIPTION
**Pull request by ebattat**
This PR is including:

1. Adding LSO support for the following databases for pod/kata/vm:
    a. PostgreSQL
    b. MariaDB
    c. MSSQL
2. Adding storage_type field indication: odf, lso, ephemeral
3. Enable PostgreSQL pod/kata/vm in functional CI.

### CODE REVIEW

File: .github/workflows/Nightly_Func_Env_CI.yml [hunk 1]
```diff
@@ -90,7 +90,7 @@ jobs:
        # continue to next job if failed
        fail-fast: false
        matrix: 
-          workload: [ 'uperf_pod', 'uperf_kata', 'uperf_vm', 'hammerdb_pod_mariadb', 'hammerdb_kata_mariadb',  'hammerdb_vm_mariadb', 'hammerdb_pod_postgres', 'hammerdb_kata_postgres', 'hammerdb_vm_postgres', 'hammerdb_pod_mssql', 'hammerdb_kata_mssql', 'hammerdb_vm_mssql', 'vdbench_pod', 'vdbench_kata', 'vdbench_vm', 'vdbench_pod_scale', 'vdbench_kata_scale', 'vdbench_vm_scale', 'clusterbuster']
+          workload: [ 'uperf_pod', 'uperf_kata', 'uperf_vm', 'hammerdb_pod_mariadb', 'hammerdb_kata_mariadb',  'hammerdb_vm_mariadb', 'hammerdb_pod_postgres', 'hammerdb_kata_postgres', 'hammerdb_vm_postgres', 'hammerdb_pod_postgres_lso', 'hammerdb_kata_postgres_lso', 'hammerdb_vm_postgres_lso', 'hammerdb_pod_mssql', 'hammerdb_kata_mssql', 'hammerdb_vm_mssql', 'vdbench_pod', 'vdbench_kata', 'vdbench_vm', 'vdbench_pod_scale', 'vdbench_kata_scale', 'vdbench_vm_scale', 'clusterbuster']
```
Comment (RobertKrawitz) [inline .github/workflows/Nightly_Func_Env_CI.yml:93]: Properly this would be done for the other databases and vdbench, but I do understand that that would add considerable time.

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py [hunk 2]
```diff
@@ -451,13 +453,15 @@ def odf_pvc_verification(self):
     @logger_time_stamp
     def delete_all(self):
         """
-        This method delete all pod in namespace
+        This method delete all pod or unbounded pv in namespace
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py:456]: "unbound".

File: benchmark_runner/common/oc/oc.py [hunk 3]
```diff
@@ -150,6 +150,17 @@ def get_worker_nodes(self):
         """
         return self.run(fr""" {self.__cli} get nodes -l node-role.kubernetes.io/worker= -o jsonpath="{{range .items[*]}}{{.metadata.name}}{{'\n'}}{{end}}" """)
 
+    def delete_available_released_pv(self):
+        """
+        This method delete available pv because it's cause a bound issue when different volumeMode is available
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:155]: What kind of issue?  This wording is not clear.

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml [hunk 4]
```diff
@@ -20,7 +42,7 @@ spec:
     name: hammerdb
     args:
       pin: {{ pin }} # enable for nodeSelector
-      pin_node: "{{ pin_node1 }}"
+      pin_node: "{{ pin_node2 }}"
```
Comment (RobertKrawitz) [inline benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml:45]: Why this change?

File: benchmark_runner/main/environment_variables.py [hunk 5]
```diff
@@ -54,10 +56,11 @@ def __init__(self):
...
         # ElasticSearch
-        self._environment_variables_dict['elasticsearch'] = EnvironmentVariables.get_env('ELASTICSEARCH', '')
-        self._environment_variables_dict['elasticsearch_port'] = EnvironmentVariables.get_env('ELASTICSEARCH_PORT', '')
-        self._environment_variables_dict['elasticsearch_user'] = EnvironmentVariables.get_env('ELASTICSEARCH_USER', '')
-        self._environment_variables_dict['elasticsearch_password'] = EnvironmentVariables.get_env('ELASTICSEARCH_PASSWORD', '')
+        # ElasticSearch
+        self._environment_variables_dict['elasticsearch'] = os.environ.get('ELASTICSEARCH', '')
+        self._environment_variables_dict['elasticsearch_port'] = os.environ.get('ELASTICSEARCH_PORT', '')
+        self._environment_variables_dict['elasticsearch_user'] = os.environ.get('ELASTICSEARCH_USER', '')
+        self._environment_variables_dict['elasticsearch_password'] = os.environ.get('ELASTICSEARCH_PASSWORD', '')
```
Comment (RobertKrawitz) [inline benchmark_runner/main/environment_variables.py:63]: Why this change?  That prevents passing these arguments on the command line.

File: benchmark_runner/main/environment_variables.py [hunk 6]
```diff
@@ -76,8 +79,11 @@ def __init__(self):
         self._environment_variables_dict['workloads'] = ['stressng_pod', 'stressng_vm', 'stressng_kata',
                                                          'uperf_pod', 'uperf_vm', 'uperf_kata',
                                                          'hammerdb_pod_mariadb', 'hammerdb_vm_mariadb', 'hammerdb_kata_mariadb',
+                                                         'hammerdb_pod_mariadb_lso', 'hammerdb_vm_mariadb_lso', 'hammerdb_kata_mariadb_lso',
                                                          'hammerdb_pod_postgres', 'hammerdb_vm_postgres', 'hammerdb_kata_postgres',
+                                                         'hammerdb_pod_postgres_lso', 'hammerdb_vm_postgres_lso', 'hammerdb_kata_postgres_lso',
                                                          'hammerdb_pod_mssql', 'hammerdb_vm_mssql', 'hammerdb_kata_mssql',
+                                                         'hammerdb_pod_mssql_lso', 'hammerdb_vm_mssql_lso', 'hammerdb_kata_mssql_lso',
                                                          'vdbench_pod', 'vdbench_kata', 'vdbench_vm',
```
Comment (RobertKrawitz) [inline benchmark_runner/main/environment_variables.py:87]: Really should do the same for vdbench.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: .github/workflows/Nightly_Func_Env_CI.yml (hunk 1 above)
Comment (ebattat) [reply .github/workflows/Nightly_Func_Env_CI.yml:93]: No, we plan to run only postgresSQL using LSO for now because run time limitation.

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py (hunk 2 above)
Comment (ebattat) [reply benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py:456]: Fixed

File: benchmark_runner/common/oc/oc.py (hunk 3 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py:155]: Fixed

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml (hunk 4 above)
Comment (ebattat) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml:45]: We have the LSO disk available only on node2.

File: benchmark_runner/main/environment_variables.py (hunk 5 above)
Comment (ebattat) [reply benchmark_runner/main/environment_variables.py:63]: Fixed

File: benchmark_runner/main/environment_variables.py (hunk 6 above)
Comment (ebattat) [reply benchmark_runner/main/environment_variables.py:87]: We can discuss it but it should be on other PR

File: .github/workflows/Nightly_Func_Env_CI.yml (hunk 1 above)
Comment (RobertKrawitz) [reply .github/workflows/Nightly_Func_Env_CI.yml:93]: OK.

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml (hunk 4 above)
Comment (RobertKrawitz) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml:45]: Then change pin_node1 in the environment.  The repository shouldn't contain configuration-specific information.

File: benchmark_runner/main/environment_variables.py (hunk 6 above)
Comment (RobertKrawitz) [reply benchmark_runner/main/environment_variables.py:86]: That's fine.

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml (hunk 4 above)
Comment (ebattat) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml:45]: We have only 1 disk for LSO and its located on in one of the nodes for cost saving, that the reason we need to pin specific node.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
Just one more change: the code in the repository should not reflect the configuration of a particular CI environment.  Use the action command line to set pin_node1 appropriately.

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml (hunk 4 above)
Comment (RobertKrawitz) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml:45]: Right.  Change the value of pin_node1 in the Github action, not in the code.

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mssql_template.yaml [hunk 7]
```diff
@@ -71,31 +112,18 @@ spec:
             value: "Y"
           - name: MSSQL_SA_PASSWORD
             value: "s3curePasswordString"
-        {%- if odf_pvc == True %}
+        securityContext:
+          allowPrivilegeEscalation: true
+      {%- if odf_pvc == True %}
```
Comment (jeniferh) [inline benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mssql_template.yaml:117]: Unless I misunderstood, the below volumeMounts/volumes are not defined at all for LSO mode?

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mariadb_template.yaml [hunk 8]
```diff
@@ -109,20 +150,7 @@ spec:
         - name: mariadb-persistent-storage
```
Comment (jeniferh) [inline benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mariadb_template.yaml:150]: Is mariadb-persistent-storage volume not needed for LSO mode?

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/postgres_template.yaml [hunk 9]
```diff
@@ -94,7 +135,7 @@ spec:
             - name: postgres-persistent-storage
```
Comment (jeniferh) [inline benchmark_runner/common/template_operations/templates/hammerdb/internal_data/postgres_template.yaml:135]: Is postgres-persistent-storage not needed for LSO mode?

**Review by jeniferh — COMMENTED** [review]
Left a few comments, I'm a little confused about some missing volumes in the Deployments when LSO will be used

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mssql_template.yaml (hunk 7 above)
Comment (ebattat) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mssql_template.yaml:117]: You right, I run LSO always with odf_pvc == True.
I have fixed it in all the database:
`{%- if storage_type == 'lso' or odf_pvc == True %} `

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mariadb_template.yaml (hunk 8 above)
Comment (ebattat) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/mariadb_template.yaml:150]: You right, I run LSO always with odf_pvc == True.
I have fixed it in all the database:
{%- if storage_type == 'lso' or odf_pvc == True %}

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/postgres_template.yaml (hunk 9 above)
Comment (ebattat) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/postgres_template.yaml:135]: You right, I run LSO always with odf_pvc == True.
I have fixed it in all the database:
{%- if storage_type == 'lso' or odf_pvc == True %}

File: benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml (hunk 4 above)
Comment (ebattat) [reply benchmark_runner/common/template_operations/templates/hammerdb/internal_data/hammerdb_vm_template.yaml:45]: I prefer to keep it as is, the change is only for the hammerdb vm workload

**Comment by ebattat** [comment]
> RobertKrawitz

I prefer not to change it in action because it will affects all other workflows.
